dynamiq/memory/backend/qdrant.py

```diff
@@ -0,0 +1,146 @@
+import uuid
+
+from qdrant_client.http import models as qdrant_models
+from qdrant_client.http.exceptions import ApiException, UnexpectedResponse
+from qdrant_client.models import FieldCondition, Filter, MatchAny, MatchValue, Range
+
+from dynamiq.components.embedders.base import BaseEmbedder
+from dynamiq.connections import Qdrant as QdrantConnection
+from dynamiq.memory.backend.base import MemoryBackend
+from dynamiq.prompts import Message
+
+
+class QdrantError(Exception):
+    """Base exception for Qdrant-related errors."""
+
+    pass
+
+
+class Qdrant(MemoryBackend):
+    """Qdrant implementation of the memory storage backend."""
+
+    name = "Qdrant"
+
+    def __init__(self, connection: QdrantConnection, embedder: BaseEmbedder, collection_name: str = "conversations"):
```

Comment:
Should we always use `index_name` instead of `collection_name`. For writers and readers, we always call it `index_name`. I would prefer to keep it consistent. 

Generally, I would prefer to keep the same names as those used by the original database terminology (e.g. `index` for weaviate, `collection` for chroma...)

Reply:
Let's keep it consistent with the our package. 

Even though I don't like this :/